active_learning/liver_tumor_al/active_learning.py

```diff
@@ -0,0 +1,607 @@
+            network.train()
+            epoch_loss = 0
+            step = 0
+            for batch_data in train_loader:
+                step += 1
+
+                inputs, labels = (
+                    batch_data["image"].to(device),
+                    batch_data["label"].to(device),
+                )
+
+                optimizer.zero_grad()
+                outputs = network(inputs)
+
+                loss_dice = dice_loss_function(outputs, labels)
+
+                loss_dice.backward()
+                optimizer.step()
+
+                # Total Loss Storage of value
+                epoch_loss += loss_dice.item()
+
+                print(f"{step}/{(len(train_ds)) // train_loader.batch_size}, " f"train_loss: {loss_dice.item():.4f}")
+
+            epoch_loss /= step
+            epoch_loss_values.append(epoch_loss)
+            writer.add_scalar("Train/Dice Loss", epoch_loss, epoch)
+            print(f"epoch {epoch + 1} average loss: {epoch_loss:.4f}")
+
+            if (epoch + 1) % args.val_interval == 0:
+                network.eval()
+                with torch.no_grad():
+                    for val_data in val_loader:
+                        val_inputs, val_labels = (
+                            val_data["image"].to(device),
+                            val_data["label"].to(device),
+                        )
+                        roi_size = (160, 160, 160)
+                        sw_batch_size = 4
+                        val_outputs = sliding_window_inference(val_inputs, roi_size, sw_batch_size, network)
+                        val_outputs = [post_pred(i) for i in decollate_batch(val_outputs)]
+                        val_labels = [post_label(i) for i in decollate_batch(val_labels)]
+                        # compute metric for current iteration
+                        dice_metric(y_pred=val_outputs, y=val_labels)
+
+                    # aggregate the final mean dice result
+                    metric = dice_metric.aggregate().item()
+                    # reset the status for next validation round
+                    dice_metric.reset()
+                    writer.add_scalar("Validation/Dice", metric, epoch)
+                    metric_values.append(metric)
+                    if metric > best_metric:
+                        best_metric = metric
+                        best_metric_epoch = epoch + 1
+                        torch.save(network.state_dict(), os.path.join(active_model_dir, "model.pt"))
+                        print("saved new best metric model")
+                    print(
+                        f"current epoch: {epoch + 1} current mean dice: {metric:.4f}"
+                        f"\nbest mean dice: {best_metric:.4f} "
+                        f"at epoch: {best_metric_epoch}"
+                    )
+
+        # Model training ends for a single active iteration
+
+        # Beginning Acquisition Strategy function
+        print("Prepping to run inference on unlabeled pool of data")
```

Comment:
Hi @finalelement ,
    I think it would be better to move this print to line 437. Because the code block between this line and line 437 does the inference on the test dataset instead of the unlabeled dataset.

Thanks,
Bin

Reply:
fixed.